pkg/synchronization/controller.go

```diff
@@ -534,6 +539,38 @@ func (c *Controller) createOrUpdateSecret(logger *zap.Logger, secretSpec *core_v
 	return true, secret, err
 }
 
+func (c *Controller) ensureCommonSecretExists(logger *zap.Logger, ns *core_v1.Namespace) (bool /* retriable */, error) {
+	_, err := c.MainClient.CoreV1().Secrets(ns.Name).Get(apik8scompute.CommonSecretName, meta_v1.GetOptions{})
+
+	// nothing to do, it exists; nothing to update
+	if err == nil {
+		logger.Sugar().Infof("Secret named %q already exists", apik8scompute.CommonSecretName)
+		return false, nil
+	}
+
+	// not found, create it
+	if api_errors.IsNotFound(err) {
+		logger.Sugar().Infof("Common secret %q does not exist, creating", apik8scompute.CommonSecretName)
+		secret := &core_v1.Secret{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:      apik8scompute.CommonSecretName,
+				Namespace: ns.Name,
+			},
+			TypeMeta: meta_v1.TypeMeta{
+				Kind:       k8s.SecretKind,
+				APIVersion: core_v1.SchemeGroupVersion.String(),
+			},
+			Type: core_v1.SecretTypeOpaque,
+		}
+
+		_, err = c.MainClient.CoreV1().Secrets(ns.Name).Create(secret)
+		return true, err
+	}
+
+	// unknown error
+	return false, err
```

Comment:
Could be retriable? e.g. in case of request timeout or some other temporary network glitch

Reply:
Updated.